bindings/node/src/mongocrypt.cc

```diff
@@ -263,656 +233,551 @@ bool MongoCrypt::setupCryptoHooks(mongocrypt_t* mongoCrypt, CryptoHooks* cryptoH
+            Array keyAltNamesArray = keyAltNames.As<Array>();
+            uint32_t keyAltNamesLength = keyAltNamesArray.Length();
             for (uint32_t i = 0; i < keyAltNamesLength; i += 1) {
-                if (Nan::Has(keyAltNamesArray, i).FromMaybe(false)) {
-                    v8::Local<v8::Object> keyAltName =
-                        Nan::To<v8::Object>(Nan::Get(keyAltNamesArray, i).ToLocalChecked())
-                            .ToLocalChecked();
-                    if (!node::Buffer::HasInstance(keyAltName)) {
+                if (keyAltNamesArray.Has(i)) {
+                    Napi::Value keyAltName = keyAltNamesArray[i];
+                    if (!keyAltName.IsBuffer()) {
                         // We should never get here
-                        Nan::ThrowTypeError("Serialized keyAltName must be a Buffer");
-                        return;
+                        throw TypeError::New(Env(), "Serialized keyAltName must be a Buffer");
                     }
 
                     std::unique_ptr<mongocrypt_binary_t, MongoCryptBinaryDeleter> binary(
-                        BufferToBinary(keyAltName));
+                        BufferToBinary(keyAltName.As<Uint8Array>()));
                     if (!mongocrypt_ctx_setopt_key_alt_name(context.get(), binary.get())) {
-                        Nan::ThrowTypeError(errorStringFromStatus(context.get()));
-                        return;
+                        throw TypeError::New(Env(), errorStringFromStatus(context.get()));
                     }
                 }
             }
         }
     }
 
     if (!mongocrypt_ctx_datakey_init(context.get())) {
-        Nan::ThrowTypeError(errorStringFromStatus(context.get()));
-        return;
+        throw TypeError::New(Env(), errorStringFromStatus(context.get()));
     }
 
-    v8::Local<v8::Object> result = MongoCryptContext::NewInstance(context.release());
-    info.GetReturnValue().Set(result);
+    return MongoCryptContext::NewInstance(Env(), std::move(context));
 }
 
-NAN_MODULE_INIT(MongoCryptContext::Init) {
-    v8::Local<v8::FunctionTemplate> tpl = Nan::New<v8::FunctionTemplate>();
-    tpl->SetClassName(Nan::New("MongoCryptContext").ToLocalChecked());
-    Nan::SetPrototypeMethod(tpl, "nextMongoOperation", NextMongoOperation);
-    Nan::SetPrototypeMethod(tpl, "addMongoOperationResponse", AddMongoOperationResponse);
-    Nan::SetPrototypeMethod(tpl, "finishMongoOperation", FinishMongoOperation);
-    Nan::SetPrototypeMethod(tpl, "nextKMSRequest", NextKMSRequest);
-    Nan::SetPrototypeMethod(tpl, "finishKMSRequests", FinishKMSRequests);
-    Nan::SetPrototypeMethod(tpl, "finalize", Finalize);
+// Store callbacks as nested properties on the MongoCrypt binding object
+// itself, and use these helpers to do so. Storing them as JS engine
+// References is a big memory leak footgun.
+Function MongoCrypt::GetCallback(const char* name) {
+    InstanceData* instance_data = Env().GetInstanceData<InstanceData>();
```

Comment:
Does fetching this have some side effect? its being reported as unused.

Reply:
Err, yeah, thanks for catching. It’s a leftover from a previous version.